src/Symfony/Component/Translation/Dumper/YamlFileDumper.php

```diff
@@ -22,11 +22,59 @@
 class YamlFileDumper extends FileDumper
 {
     /**
+     * Dump messages as tree, e.g. this
+     *   'foo.bar1' => 'test1',
+     *   'foo.bar2' => 'test2'
+     *
+     * dumps as follows:
+     *   foo:
+     *     bar1: test1
+     *     bar2: test2
+     *
+     * @var bool
+     */
+    private $asTree = true;
+
+    /**
+     * The level where you switch to inline YAML.
+     * Do not switch to inline YAML by default.
+     *
+     * @var int
+     */
+    private $inline = 999;
+
+    /**
+     * Sets asTree flag
+     * 
+     * @param bool $asTree
+     */
+    public function setDumpAsTree($asTree)
```

Comment:
I think the BC break is fine by looking at implementations of this outside Symfony: http://packanalyst.com/class?q=Symfony\Component\Translation\Dumper\FileDumper

Drupal and Prezent are extending the YamlFileDumper to implement this feature, meaning their upgrade path will be easy: switch to the core one.

And then, we could even submit PRs ourselves to Sulu and Kdyby to update their implementations